src/main/java/gov/nasa/jpl/mbee/mdk/ems/actions/CommitOrgAction.java

```diff
@@ -0,0 +1,148 @@
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON 
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE 
+ * OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE 
+ * POSSIBILITY OF SUCH DAMAGE.
+ ******************************************************************************/
+package gov.nasa.jpl.mbee.mdk.ems.actions;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.nomagic.magicdraw.annotation.Annotation;
+import com.nomagic.magicdraw.annotation.AnnotationAction;
+import com.nomagic.magicdraw.core.Application;
+import com.nomagic.magicdraw.core.Project;
+import gov.nasa.jpl.mbee.mdk.MDKPlugin;
+import gov.nasa.jpl.mbee.mdk.api.incubating.MDKConstants;
+import gov.nasa.jpl.mbee.mdk.docgen.validation.IRuleViolationAction;
+import gov.nasa.jpl.mbee.mdk.docgen.validation.RuleViolationAction;
+import gov.nasa.jpl.mbee.mdk.ems.MMSUtils;
+import gov.nasa.jpl.mbee.mdk.ems.ServerException;
+import gov.nasa.jpl.mbee.mdk.json.JacksonUtils;
+import gov.nasa.jpl.mbee.mdk.lib.MDUtils;
+import org.apache.http.client.utils.URIBuilder;
+
+import javax.swing.*;
+import java.awt.event.ActionEvent;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.Collection;
+
+public class CommitOrgAction extends RuleViolationAction implements AnnotationAction, IRuleViolationAction {
+
+    public static final String DEFAULT_ID = CommitOrgAction.class.getSimpleName();
+    public static final String COMMIT_MODEL_DEFAULT_ID = DEFAULT_ID + "_Commit_Org";
+    private final Project project;
+
+    public CommitOrgAction(Project project) {
+        this(project, false);
+    }
+
+    public CommitOrgAction(Project project, boolean isDeveloperAction) {
+        super(DEFAULT_ID, "Commit Org" + (isDeveloperAction ? " [DEVELOPER]" : ""), null, null);
+        this.project = project;
+    }
+
+    @Override
+    public boolean canExecute(Collection<Annotation> arg0) {
+        return false;
+    }
+
+
+    @Override
+    public void execute(Collection<Annotation> annos) {
+    }
+
+    @Override
+    public void actionPerformed(ActionEvent e) {
+        commitAction();
+    }
+
+    public String commitAction() {
+        if (!MDUtils.isDeveloperMode()) {
+            return null;
+        }
+
+        // check for existing org
+        URIBuilder requestUri = MMSUtils.getServiceOrgsUri(project);
+        if (requestUri == null) {
+            return null;
+        }
+
+        JFrame selectionDialog = new JFrame();
+        String org = JOptionPane.showInputDialog(selectionDialog, "[DEVELOPER] Input MMS org below");
+        if (org == null || org.isEmpty()) {
+            Application.getInstance().getGUILog().log("[ERROR] Unable to commit org without name.");
```

Comment:
Consider cancel/close case. It could be an intentional cancellation, so we should not give the impression that we tried to commit. We should also specify what we are doing i.e. "Org commit cancelled."